c_binding/libsg.c

```diff
@@ -114,6 +114,15 @@ struct _sg_t10_vpd83_header {
     uint16_t page_len_be;
 };
 
+struct _sg_t10_vpd80_header {
+    uint8_t data_msb : 5;
+    uint8_t naa_type : 3;
+    uint8_t page_code;
+    uint8_t rsvd;
+    uint8_t page_len_be;
+    uint8_t serial_num[_SG_T10_SPC_SN_MAX_LEN];
+};
+
```

Comment:
The `struct _sg_t10_vpd80_header` here is incorrect. According to SCSI SPC-4 rev37 and SPC-5 rev7, it should be:
```
struct _sg_t10_vpd80_header {
    uint8_t dev_type : 5;
    uint8_t qualifier : 3;
    uint8_t page_code;
    uint16_t page_len_be;
};
```
And you should not include `uint8_t serial_num[_SG_T10_SPC_SN_MAX_LEN]` in this struct as `serial_num` is not fixed length.